src/unstated.js

```diff
@@ -103,8 +85,10 @@ export class Subscribe<Containers: ContainersType> extends React.Component<
     });
   }
 
-  onUpdate: Listener = cb => {
-    this.setState(DUMMY_STATE, cb);
+  onUpdate: Listener = () => {
+    return new Promise(resolve => {
```

Comment:
why not just `return this.setState(DUMMY_STATE);`?

Reply:
because then we dont wait for the callback